### PR TITLE
Integrate UART Library

### PR DESCRIPTION
**Pull request by jack-greenberg**

### CODE REVIEW

File: libs/uart/uart.c [hunk 1]
```diff
@@ -0,0 +1,54 @@
+#include "libs/uart/api.h"
+#include <avr/io.h>
+
+void uart_init(uint16_t baudrate, bool use_interrupt) {
+    // TODO: Implement
+    (void)baudrate;
+    (void)use_interrupt;
+
+    LINCR |= (1 << LENA) | 0x7; // Enable UART
```
Comment (mpatil99) [inline libs/uart/uart.c:9]: Datasheet line num?

File: libs/uart/uart.c [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+
+char uart_getchar(void) {
+    loop_until_bit_is_clear(LINSIR, LBUSY);
+    char c = LINDAT;
+    return c;
+}
+
+void uart_putchar(char c) {
+    loop_until_bit_is_clear(LINSIR, LBUSY);
+    LINDAT = c;
```
Comment (mpatil99) [inline libs/uart/uart.c:21]: Datasheet line num?

File: libs/uart/uart.c [hunk 3]
```diff
@@ -0,0 +1,54 @@
+#include "libs/uart/api.h"
+#include <avr/io.h>
+
+void uart_init(uint16_t baudrate, bool use_interrupt) {
+    // TODO: Implement
```
[the remark this answers is not in the document]
Comment (mpatil99) [reply libs/uart/uart.c:5]: Maybe just a single link up top with URL + page range

**Review by mpatil99 — APPROVED** [review]
Looks good overall. Mostly just nits

File: examples/blinky/main.c [hunk 4]
```diff
@@ -1,14 +1,18 @@
+#include "libs/uart/api.h"
 #include <avr/io.h>
 #include <util/delay.h>
 
 #define LED0 (PD6)
 
 int main(void) {
     DDRD |= _BV(LED0);
```
Comment (awenstrup) [inline examples/blinky/main.c:8]: Can't wait til the GPIO library is a thing so I don't have to look at this anymore lol

File: libs/uart/uart.c [hunk 5]
```diff
@@ -0,0 +1,67 @@
+/* Atmega 16M1 Uart Library Implementation
+All page numbers refer to
+http://ww1.microchip.com/downloads/en/DeviceDoc/Atmel-7647-Automotive-Microcontrollers-ATmega16M1-32M1-64M1-32C1-64C1_datasheet.pdf
+*/
+#include "libs/uart/api.h"
+#include <avr/io.h>
+
+void uart_init(uint16_t baudrate, bool use_interrupt) {
+    // TODO: Implement
+    (void)use_interrupt;
```
Comment (awenstrup) [inline libs/uart/uart.c:10]: Can you provide some background on this in a comment? I haven't worked with uart in a while so maybe I'm just rusty but I have no clue when I'd want to set this true or false.

**Pull request merged**